### PR TITLE
#7, #8: Added Dockerfiles and docker-compose + flask and mongodb containers

### PR DESCRIPTION
**Pull request by mma01us**
### Testing:
- Test with ```docker compose up```.
- You should be able to visit ```http://127.0.0.1:5000``` in your browser and view the webpages (**will be blank since database is empty**).
- You can modify and check the database with a tool like MongoDB Compass - connect with ```mongodb://127.0.0.1:27017```

### Notes:
- In each container, the host ```127.0.0.1``` is mapped to ```172.17.0.x``` addresses
- As the connection string for the database is different when run via Docker containers, the DB_URL environment variable is used as a workaround. ```__init__.py``` has been modified to reflect this and defaults to the ```127.0.0.1``` ip if DB_URL is not present. The compose file will set DB_URL accordingly.

### CODE REVIEW

File: machine-learning-client/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+WORKDIR /machine-learning-client
+
+# Add files from machine to Docker image
+ADD . /machine-learning-client
+
+# Install dependencies and packages
+RUN pip install -r requirements.txt
+
+# TODO: run ml client on 127.0.0.1:6000
+CMD ["python3", "-m", "flask", "--app=flaskr", "--debug", "run", "--host=0.0.0.0"]
```
Comment (bxw201) [inline machine-learning-client/Dockerfile:14]: Does the machine learning client also involve running a flask app?

Comment (mma01us) [reply machine-learning-client/Dockerfile:14]: This is untested and currently just a placeholder (as noted by TODO). Currently, docker compose only handles the Flask and MongoDB setup. The ML Client Dockerfile will be updated and added later.

Comment (mma01us) [reply machine-learning-client/Dockerfile:14]: Update: We won't be dockerizing the ML client so the Dockerfile in the ML client directory will be removed.

**Review by bxw201 — APPROVED** [review]
Thanks for the testing and notes. LGM

**Pull request merged**